frontend/src/app/utils/timeline/timeline-creator.ts

```diff
@@ -0,0 +1,296 @@
+    }
+
+    private getExceedingBlocks(exceeding: Task[][]): TasksBlock[] {
+        let exceedingTaskBlocks: Task[][] = [];
+        exceeding = exceeding.sort(
+            (firstBlock, secondBlock) => this.getBlockStart(firstBlock).getTime() - this.getBlockStart(secondBlock).getTime()
+        );
+        let lastBlockIndex = 0;
+        exceedingTaskBlocks = [exceeding[0]];
+        for (let i = 1; i < exceeding.length; i++) {
+            if (this.getBlockEnd(exceedingTaskBlocks[lastBlockIndex]) > this.getBlockStart(exceeding[i])) {
+                exceedingTaskBlocks[lastBlockIndex] = exceedingTaskBlocks[lastBlockIndex].concat(exceeding[i]);
+            } else {
+                exceedingTaskBlocks.push(exceeding[i]);
+                lastBlockIndex++;
+            }
+        }
+        return exceedingTaskBlocks.map(tasks => ({
+            tasks,
+            expanded: false,
+            startColumn: this.getColumn(this.getBlockStart(tasks)),
+            endColumn: this.getColumn(this.getBlockEnd(tasks))
+        }));
+    }
+
+    private addToExceeding(exceeding: Task[][], task: Task): void {
+        let inserted = false;
+        for (let i = 0; i < exceeding.length && !inserted; i++) {
+            let block = exceeding[i];
+            if (intercept(task, { start: this.getBlockStart(block), end: this.getBlockEnd(block)})) {
+                block.push(task);
+                inserted = true;
+            }
+        }
+        if (!inserted) {
+            exceeding.push([task]);
+        }
+    }
+
+    private getTimelineBoundaries(currentDate: Date, scale: TimelineScale, pageNumber: number): TimelineSection {
+        switch (scale) {
+        case TimelineScale.Day:
+            return {
+                start: startOfDay(addDays(currentDate, -1 * pageNumber / 3)),
+                end: endOfDay(addDays(currentDate, pageNumber / 3))
+            };
+        case TimelineScale.Week:
+            return {
+                start: startOfWeek(addWeeks(currentDate, -1 * pageNumber / 2)),
+                end: endOfWeek(addWeeks(currentDate, pageNumber / 2))
+            };
+        case TimelineScale.Month:
+            return {
+                start: startOfMonth(addMonths(currentDate, -1 * pageNumber / 2)),
+                end: endOfMonth(addMonths(currentDate, pageNumber / 2))
+            };
+        case TimelineScale.Year:
+            return {
+                start: startOfMonth(addYears(currentDate, -1 * pageNumber / 2)),
+                end: endOfMonth(addYears(currentDate, pageNumber / 2))
+            };
+        default:
+            throw new Error('Invalid scale');
+        }
+    }
+
+    private getTimelineSections(boundaries: TimelineSection, scale: TimelineScale) {
+        if (scale !== TimelineScale.Year) {
+            return this.getTimelineSubsections(boundaries, scale + 1);
+        }
+        let sectionHours = hoursInDay * (daysInYear + 2);
```

Comment:
Use const when you are not going to reassign variables. It is more strict. When someone will modify this method and add reasigning, dev think why is it const and is he really needs to reassign.

Reply:
Fixed and added such rule to the linter